plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/WarPublisher.java

```diff
@@ -31,13 +31,14 @@
 import org.eclipse.core.runtime.OperationCanceledException;
 import org.eclipse.core.runtime.Path;
 import org.eclipse.core.runtime.SubMonitor;
-import org.eclipse.jst.server.core.IWebFragmentModule;
+import org.eclipse.jst.server.core.IJ2EEModule;
+import org.eclipse.jst.server.core.IUtilityModule;
 import org.eclipse.wst.server.core.IModule;
 import org.eclipse.wst.server.core.ServerUtil;
 import org.eclipse.wst.server.core.model.IModuleResource;
 import org.eclipse.wst.server.core.model.ModuleDelegate;
 import org.eclipse.wst.server.core.util.ModuleFile;
-import org.eclipse.wst.server.core.util.PublishHelper;
+import org.eclipse.wst.server.core.util.PublishUtil;
 
 /**
  * Writes a WAR file of a project, or the exploded contents of it to a destination directory.
```

Comment:
Looks like it just writes the exploded contents, not the WAR file?

Reply:
This is not wrong, since the Javadoc is for the class, which has methods to create both a WAR file or an exploded directory.